ext/standard/tests/file/bug28790.cache.phpt

```diff
@@ -0,0 +1,53 @@
+--TEST--
+Bug #28790: Add php.ini option to disable stat cache (with cache)
+--CREDITS--
+jnoll <jnoll [at] prim [dot] hu>
+Kevin Lyda <kevin [at] lyda [dot] ie>
+--FILE--
+<?php
+
+$php = '"'.getenv('TEST_PHP_EXECUTABLE').'"';
+$phpfile = getenv('TEST_PHP_EXECUTABLE');
+$impossiblefile = $phpfile.DIRECTORY_SEPARATOR.'bug28790.impossible';
+$testfile = __DIR__.DIRECTORY_SEPARATOR.'bug28790.file';
+
+function all_the_stats($filename, $message) {
+    if (@lstat($filename)) {
+        print("lstat: $message.\n");
+    }
+    if (@stat($filename)) {
+        print("stat: $message.\n");
+    }
+}
+
+// Windows can use / for dir separators, so let's do that.
+$qtestfile = str_replace("\\", "/", "$testfile");
+passthru($php.' -n -r "touch(\\"'.$qtestfile.'\\");"');
+all_the_stats("$testfile", "testfile exists");
+passthru($php.' -n -r "unlink(\\"'.$qtestfile.'\\");"');
+all_the_stats("$testfile", "testfile exists (it shouldn't)");
+if (!@stat("$impossiblefile")) {
+    print("stat impossiblefile does not exist.\n");
+}
+all_the_stats("$testfile", "testfile exists (it shouldn't)");
+if (is_file("$phpfile")) {
```

Comment:
What are these checks for the existence of the PHP binary for?

Reply:
The idea is to exercise the cache (or lack of cache).  And I'm using the php binary to mess with the file outside the current process w/o using system tools which would be different on various platforms.

Reply:
I understand that, but why do we need is_file and lstat checks on the PHP binary?

Reply:
Oh, and checking for the php binary is an easy way to reset the cache (only the last good stat is cached) and I know the php binary exists.

Reply:
Oh... I see.  Just a sec...

Reply:
OK, fixed the code, reran the tests just to clarify things.  Right, there are two stat caches.  One for stat calls and one for lstat calls. I'm doing things to invalidate the stat cache first and then the lstat cache.